src/main/bootstrap/mod.rs

```diff
@@ -39,6 +39,9 @@ const SERVICE_DISCOVERY_TIMEOUT_SEC: u64 = 1;
 const BOOTSTRAP_TIMER_ID: u8 = 0;
 const SERVICE_DISCOVERY_TIMER_ID: u8 = BOOTSTRAP_TIMER_ID + 1;
 const MAX_CONTACTS_EXPECTED: usize = 1500;
+/// Bootstrap will use up to max this amount of cached peers. We don't want to exhaust socket
```

Comment:
Neither this comment nor the impl is correct. We do want to go through the whole collection of everyone in the cache and in hardcoded contact list and via serv-disc. Just that we want to do max 10 at at time in parallel. It was always about parallel bootstrap limit.

So the last point [here](https://github.com/maidsafe/crust/issues/1106) is:
> Limit parallel connections during bootstrap to prevent from exhausting system sockets.

So the comment should indicate that we try this number of connection attempts in parallel at any given time. When there's a failure, take the next one up from the list and so on until the whole combined list is exhausted.